packages/contractkit/src/identity/offchain/schema-utils.ts

```diff
@@ -55,13 +65,18 @@ export class SimpleSchema<DataType> {
   deserialize(buf: Buffer): Result<DataType, SchemaErrors> {
     return deserialize(this.type, buf)
   }
+  async sign(data: DataType) {
```

Comment:
Does this need to be a public function?

Reply:
In general `SimpleSchema` seems to expose quite a fair number of methods. As the schema is the main interface for developers, it might make sense to slim the interface down to what they would actually use? I.e. just read + write?